gateway/dataimpl/server_v1/kvserver.go

```diff
@@ -85,6 +85,10 @@ func (s *KvServer) Get(ctx context.Context, in *kv_v1.GetRequest) (*kv_v1.GetRes
 			isFullDocFetch = true
 		}
 
+		if len(opts.Key) > 250 {
```

Comment:
Is this definitely right? Based on https://github.com/couchbase/gocbcorex/blob/master/memdx/packetwriter.go#L32-L34 that'd suggest a much longer key is allowed?

Reply:
That's where I got the original max len I used from. But Brett sent me this: https://docs.couchbase.com/server/current/learn/clusters-and-availability/size-limitations.html

And explained: 

"The history here is that key_length used to be a uint16 in the protocol.  Then we added framing extras and co-opted one of its bytes for that, since key length was already restricted anyways.  Leading to only a 255 bytes length.  We then added collections support which LEB encodes into the key, consuming up to 5 bytes (for a LEB encoded uint32), leaving only 250 bytes remaining for the key."

Reply:
Ahh cool. Have you tried with a key length of 251 and 256 to see what happens?

Reply:
I haven't tried sending a key of those lengths but I can give it a go

Reply:
Maybe just change `s.tooLongDocId(),` into `s.longDocId(250)`, and then add one other tests which checks that `s.longDocId(249)` works, while the existing test would check that 250 fails.

Reply:
Sounds reasonable, will get on that. Was @chvck asking what happens if we have those key lengths but remove the stella-gateway level check? Because that should return an error right? But it will just be uninformative

Reply:
> Ahh cool. Have you tried with a key length of 251 and 256 to see what happens?

@chvck tried with key len 251 and 256 and both return an UNKNOWN_ERROR status code